x-pack/plugins/apm/server/routes/metrics/by_agent/shared/memory/index.ts

```diff
@@ -62,23 +62,27 @@ export const percentSystemMemoryUsedScript = {
 export const percentCgroupMemoryUsedScript = {
   lang: 'painless',
   source: `
-    /*
-      When no limit is specified in the container, docker allows the app as much memory / swap memory as it wants.
-      This number represents the max possible value for the limit field.
-    */
-    double CGROUP_LIMIT_MAX_VALUE = 9223372036854771712L;
+    if(doc.containsKey('${METRIC_CGROUP_MEMORY_LIMIT_BYTES}') && doc.containsKey('${METRIC_CGROUP_MEMORY_USAGE_BYTES}')){
```

Comment:
What about `doc['${METRIC_SYSTEM_TOTAL_MEMORY}']` ?

Reply:
ah I was trying different things and forgot to add the one that was causing issue 🙈 
good catch!

Reply:
I can get rid of `METRIC_CGROUP_MEMORY_LIMIT_BYTES ` as it's already checking if it exists in this line https://github.com/elastic/kibana/pull/145348/files/41f4588a1e4a64e71e909d303d284cb8eaece403#diff-741ca25328532c18284b64412f95043bb968f82b1ca17fc64193317d41337460R75

Reply:
@kpatticha can you reach out to the agents or server devs to figure out which fields are available in what scenarios? I'm hoping we can figure out something like "if METRIC_CGROUP_MEMORY_LIMIT_BYTES is available, then METRIC_CGROUP_MEMORY_USAGE_BYTES must be too, and if it's not available, we should use METRIC_SYSTEM_TOTAL_MEMORY, which is always available". Doesn't necessarily have to happen in this PR but I feel like with each change we understand how memory usage is recorded less.

Reply:
sure I can try that, it will be really useful 👍

Reply:
I started gathering information on how system memory work [here](https://github.com/elastic/kibana/issues/145621) but it should not block this PR